src/Run.hs

```diff
@@ -123,11 +137,46 @@ getAddDistArgs = do
 isSuccess :: Summary -> Bool
 isSuccess s = sErrors s == 0 && sFailures s == 0
 
-doctestWithOptions :: Bool -> Bool -> Bool -> [String] -> IO Summary
-doctestWithOptions fastMode preserveIt verbose args = do
-
-  -- get examples from Haddock comments
-  modules <- getDocTests args
-
-  Interpreter.withInterpreter args $ \repl -> withCP65001 $ do
-    runModules fastMode preserveIt verbose repl modules
+collectLocations :: Module [Located DocTest] -> [String]
+collectLocations (Module{moduleSetup,moduleContent}) =
+  nubOrd (concatMap (mapMaybe go) (maybeToList moduleSetup <> moduleContent))
+ where
+  go :: Located DocTest -> Maybe String
+  go (Located (Location l _) _) = Just l
+  go _ = Nothing
+
+stripLocalDirs :: [Module [Located DocTest]] -> [String] -> IO [String]
+stripLocalDirs mods args_ = do
+  modLocs <- mapM canonicalizePath (concatMap collectLocations mods)
+  let modDirs = map dropFileName modLocs
+  go modDirs args_
+ where
+  go _modDirs [] = pure []
+  go modDirs (arg@('-':'i':dir0):args) = do
+    dir1 <- canonicalizePath dir0
+    if any (dir1 `isPrefixOf`) modDirs
+    then go modDirs args
+    else (arg:) <$> go modDirs args
+  go modDirs (arg:args) = (arg:) <$> go modDirs args
+
+doctestWithOptions :: Bool -> Bool -> Bool -> Bool -> Int -> Bool -> [String] -> IO Summary
+doctestWithOptions fastMode preserveIt verbose isolate nThreads usePackageDb args0 = do
+
+  -- get examples and properties from Haddock comments
+  modules <- getDocTests args0
+
+  -- Strip all local directories from -i flags to prevent GHCi from loading
+  -- from local module. Force it to load the modules from the package db
+  -- instead.
+  args1 <- if usePackageDb then stripLocalDirs modules args0 else pure args0
```

Comment:
Query on silently ignoring options versus erroring and explaining the incompatibility.

Either way, this is testable and should be tested.